src/irmin-layers/irmin_layers_intf.ml

```diff
@@ -28,6 +28,38 @@ module type S = sig
     ?recovery:bool ->
     repo ->
     unit Lwt.t
+  (** [freeze ?min ?max ?squash ?copy_in_upper ?min_upper ?recovery t] launches
+      an asynchronous freezing operation on the repo [t] to reduce the size of
+      the upper layer.
+
+      Let [o] be the set of objects (i.e. commits, nodes and contents) reachable
+      from the [max] commits and bounded by the [min] commits. During the
+      freeze, all objects in [o] are copied to the lower layer, if there is one.
+      Setting [squash] to [true] is equivalent to setting [min] to [max]. [min]
```

Comment:
maybe we don't need that extra `squash` argument if it's equivalent to `max = min`.

Reply:
Indeed, https://github.com/mirage/irmin/blob/master/src/irmin-pack/irmin_pack_layers.ml#L836

Reply:
Let's remove the extra `squash` argument, but we can do that in a different PR if you prefer.

Reply:
👍